src/flb_upstream.c

```diff
@@ -791,10 +756,44 @@ struct flb_connection *flb_upstream_conn_get(struct flb_upstream *u)
     }
 
     /*
-     * There are no keepalive connections available or keepalive is disabled
-     * so we need to create a new one.
+     * If no reusable connection is found, then check the total connection limit
+     * and create a new connection if allowed.
      */
+
     if (conn == NULL) {
+    /* If the upstream is limited by max connections, check current state */
+        if (u->base.net.max_worker_connections>0) {
+            /*
+             * Connections are linked to one of these lists:
+             *
+             *  - av_queue  : connections ready to be used (available)
+             *  - busy_queue: connections that are busy (someone is using them)
+             *  - drop_queue: connections in the cleanup phase (to be drop)
```

Comment:
good time to fix the wrong name, the queue is called `destroy_queue` instead of `drop_queue`